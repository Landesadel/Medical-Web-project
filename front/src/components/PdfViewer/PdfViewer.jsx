import React, {useEffect, useState} from 'react';
import { Document, Page,pdfjs  } from  'react-pdf';
import {base64} from "../../services/base64.service";
import styles from './PdfViewer.module.scss';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import DownloadPdfLink from "../DownloadPdfLink/DownloadPdfLink";
import {useAuth} from "../../hooks/useAuth";
pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`

const left = <FontAwesomeIcon icon={faChevronLeft} />;
const right = <FontAwesomeIcon icon={faChevronRight} />;

const PdfViewer = ({url}) => {
    const {user}=useAuth();
    const [file,setFile]=useState();
    const [numPages, setNumPages] = useState(null);
    const [pageNumber, setPageNumber] = useState(1);

    base64.getBase64(url,setFile).catch();

    function onDocumentLoadSuccess({ numPages }) {
        setNumPages(numPages);
        setPageNumber(1);
    }

    function changePage(offset) {
        setPageNumber(prevPageNumber => prevPageNumber + offset);
    }

    function previousPage() {
        changePage(-1);
    }

    function nextPage() {
        changePage(1);
    }



    return  <>

        <Document
            file={file}
            // options={{ workerSrc: "/pdf.worker.js" }}
            onLoadSuccess={onDocumentLoadSuccess}
        >
            <Page pageNumber={pageNumber}/>
        </Document>
        <div className={styles.pdfButtons}>
            <p>
                {pageNumber || (numPages ? 1 : "--")} of {numPages || "--"}
            </p>
            <button type="button" disabled={pageNumber <= 1} onClick={previousPage}>
                {left}
            </button>
            <button
                type="button"
                disabled={pageNumber >= numPages}
                onClick={nextPage}
            >
                {right}
            </button>
        </div>
        {user? (
            <DownloadPdfLink url={url} text={`Скачать файл PDF`} style={styles.link}/>
        ) : ''
        }

    </>
};

export  default PdfViewer;