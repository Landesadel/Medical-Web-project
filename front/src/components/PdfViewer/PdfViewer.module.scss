$max-tablet-width: '991px';
$max-mobile-width: '768px';

@mixin tablet {
  @media(max-width: $max-tablet-width) {
    @content;
  }
}

@mixin mobile {
  @media(max-width: $max-mobile-width) {
    @content;
  }
}


.pdfButtons {
  width: 100%;
  max-width: 120px;
  margin: 10px auto;
  display: flex;
  //flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  transform: translateX(-50%, 50%);


  button {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 20px;
    background: #3f51b5;
    padding: 4px;
    color: white;
    border: none;
    font-size: 14px;
    cursor: pointer;
    box-shadow: 2px 2px 2px 1px #ccc;

  }
}

.link {
  display: block;
  width: 100%;
  max-width: 300px;
  margin: 10px auto;
  transform: translateX(-50%, 50%);
  font-size: 15px;
  text-decoration: underline;
  color: rgba(126, 131, 166, 0.8);
  transition: font-size 0.3s linear;
  opacity: 1.0;

  &:hover {
    transition: all 0.3s linear;
    opacity: 0.7;
  }
}

//.back {
//  background: linear-gradient(to right, rgba(255, 228, 230, 0.2), rgba(204, 251, 241, 0.4));
//}
