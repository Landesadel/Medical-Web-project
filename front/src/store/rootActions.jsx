import * as userActions from './user/user.actions';
import * as forumActions from './forum/forum.actions';

export const allActions = {
	...userActions,
	...forumActions,
};
