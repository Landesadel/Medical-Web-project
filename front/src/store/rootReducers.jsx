import { reducer as userReducer } from './user/user.slice';
import { reducer as forumReducer } from './forum/forum.slice';
import { reducer as toastrReducer } from 'react-redux-toastr';

export const reducers = {
	user: userReducer,
	toastr: toastrReducer,
	forum: forumReducer,
};
