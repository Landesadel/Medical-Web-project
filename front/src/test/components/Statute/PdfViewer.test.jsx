import React from 'react';
import { render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import PdfViewer from "../../../components/PdfViewer/PdfViewer";
import {Provider} from "react-redux";
import {store} from "../../../store/store";
import {MemoryRouter} from "react-router-dom";


describe('PdfViewer', () => {
    it('render pdf-viewer with file',() => {
        const snapshot=render(
            <Provider store={store}>
                <MemoryRouter>
            <PdfViewer url='statute2022.pdf'/>
                </MemoryRouter>
            </Provider>
        );

  expect(snapshot).toMatchSnapshot();
    });

});
