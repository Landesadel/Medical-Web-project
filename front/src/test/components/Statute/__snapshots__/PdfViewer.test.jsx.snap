// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfViewer render pdf-viewer with file 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="react-pdf__Document"
      >
        <div
          class="react-pdf__message react-pdf__message--no-data"
        >
          No PDF file specified.
        </div>
      </div>
      <div
        class="pdfButtons"
      >
        <p>
          1
           of 
          --
        </p>
        <button
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-left "
            data-icon="chevron-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right "
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="react-pdf__Document"
    >
      <div
        class="react-pdf__message react-pdf__message--no-data"
      >
        No PDF file specified.
      </div>
    </div>
    <div
      class="pdfButtons"
    >
      <p>
        1
         of 
        --
      </p>
      <button
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left "
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right "
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
